src/channel.rs

```diff
@@ -524,6 +524,20 @@ impl Channel {
             internal: false,
             arguments: FieldTable::new(),
         };
+        self.exchange_declare_passive_advanced(type_, exchange, options)
+    }
+
+    /// Passively  declare an exchange named `exchange` with the given type and options.
+    ///
+    /// If the server cannot declare the exchange (e.g., if the exchange already exists with a
+    /// different type or options that conflict with `options`), it will close this channel.
```

Comment:
It's been quite a while since I've dug into the AMQP spec, so I could definitely be wrong here, but this doesn't match my understanding of how passive declarations work. From https://www.rabbitmq.com/amqp-0-9-1-reference.html:

> [bit](https://www.rabbitmq.com/amqp-0-9-1-reference.html#domain.bit) passive
> 
> If set, the server will reply with Declare-Ok if the exchange already exists with the same name, and raise an error if not. The client can use this to check whether an exchange exists without modifying the server state. When set, all other method fields except name and no-wait are ignored. A declare with both passive and no-wait has no effect. Arguments are compared for semantic equivalence.

"All other method fields" should include both `type` and anything in `options`, right? Meaning the passive declaration won't detect conflicts if there are differences.